docs/topics/OAuth2.md

```diff
@@ -56,68 +41,280 @@ Scopes provide access to certain resources of a user's account. Your API client
+4. Bots have an entirely separate set of [Rate Limits](#DOCS_RATE_LIMITS/rate-limits).
+
+### Bot Authorization Flow
+
+Bot authorization is a special server-less and callback-less OAuth2 flow that makes it easy for users to add bots to guilds. The URL you create looks similar to what we use for full stack implementation:
+
+###### URL Example
+
+```
+https://discordapp.com/api/oauth2/authorize?client_id=157730590492196864&scope=bot&permissions=1
 ```
-https://discordapp.com/api/oauth2/authorize?client_id=157730590492196864&scope=webhook.incoming&redirect_uri=https%3A%2F%2Fnicememe.website&response_type=code
+
+In the case of bots, the `scope` parameter should be set to `bot`. There's also a new parameter, `permissions`, which is an integer corresponding to the [permission calculations](#DOCS_PERMISSIONS/permissions-bitwise-permission-flags) for the bot. You'll also notice the absence of `response_type` and `redirect_uri`. Bot authorization does not require these parameters because there is no need to retrieve the user's access token.
+
+When the user navigates to this page, they'll be prompted to add the bot to a guild in which they have proper permissions. On acceptance, the bot will be added. Super easy!
+
+If your bot is super specific to your private clubhouse, or you just don't like sharing, you can leave the `Public Bot` option unchecked in your application's settings. If unchecked, only you can add the bot to guilds. If marked as public, anyone with your bot's URL can add it to guilds in which they have proper permissions.
+
+### Advanced Bot Authorization
+
+Enterprising devs can extend the bot authorization functionality. You can request additional scopes outside of `bot`, which will prompt a continuation into a complete [authorization code grant flow](#DOCS_OAUTH2/authorization-code-grant) and add the ability to request the user's access token. If you request any scopes outside of `bot`, `response_type` is again mandatory; we will also automatically redirect the user to the first uri in your application's registered list unless `redirect_uri` is specified.
+
+A full OAuth2 [authorization code grant flow](#DOCS_OAUTH2/authorization-code-grant) can also be required by ticking "Require OAuth2 Code Grant" in your application's settings. If, for instance, your bot needs to store a relationship between third-party websites and guilds to which it was added, requiring the OAuth2 code grant affords you the ability to get extra data during the token exchange. When receiving the access code on redirect, there will be additional querystring parameters of `guild_id` and `permissions`. The `guild_id` parameter should only be used as a hint as to the relationship between your bot and a guild. To be sure, enable the OAuth2 code grant requirement. When you retrieve the user's access token, you'll also receive information about the guild to which your bot was added:
```

Comment:
> When receiving the access code on redirect, there will be additional querystring parameters of `guild_id` and `permissions`. The `guild_id` parameter should only be used as a hint as to the relationship between your bot and a guild. To be sure, enable the OAuth2 code grant requirement.

does not really fit into the middle of this paragraph lol.. i would pull this out to before the paragraph, then go into the full code grant option separately.